File: go-backend/internal/model/scene_prompt.go
// Package model 场景提示词实体模型
// @author <a href="https://codefather.cn">编程导航学习圈</a>
package model

import (
	"time"
)

type ScenePrompt struct {
	ID             string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	SceneID        string    `gorm:"column:sceneId;type:varchar(36);not null;index:idx_scene" json:"sceneId"`
	UserID         int64     `gorm:"column:userId;type:bigint;not null;index:idx_user" json:"userId"`
	PromptIndex    int       `gorm:"column:promptIndex;type:int;not null;index:idx_scene" json:"promptIndex"`
	Title          string    `gorm:"column:title;type:varchar(200);not null" json:"title"`
	Content        string    `gorm:"column:content;type:text;not null" json:"content"`
	Difficulty     string    `gorm:"column:difficulty;type:varchar(20)" json:"difficulty"`
	Tags           string    `gorm:"column:tags;type:json" json:"tags"`
	ExpectedOutput string    `gorm:"column:expectedOutput;type:text" json:"expectedOutput"`
	CreateTime     time.Time `gorm:"column:createTime;type:datetime;not null;default:CURRENT_TIMESTAMP;autoCreateTime" json:"createTime"`
	UpdateTime     time.Time `gorm:"column:updateTime;type:datetime;not null;default:CURRENT_TIMESTAMP;autoUpdateTime" json:"updateTime"`
	IsDelete       int       `gorm:"column:isDelete;type:tinyint;not null;default:0" json:"-"`
}

func (ScenePrompt) TableName() string {
	return "scene_prompt"
}

File: python-backend/app/models/scene_prompt.py
"""
场景提示词数据库模型
@author <a href="https://codefather.cn">编程导航学习圈</a>
"""
from datetime import datetime
from sqlalchemy import Column, String, BigInteger, Text, Integer, DateTime
from sqlalchemy.orm import declarative_base

Base = declarative_base()


class ScenePrompt(Base):
    """
    场景提示词表模型
    """
    __tablename__ = 'scene_prompt'

    id = Column(String(36), primary_key=True, comment='提示词唯一标识')
    scene_id = Column('sceneId', String(36), nullable=False, comment='场景ID')
    user_id = Column('userId', BigInteger, nullable=False, comment='用户ID')
    prompt_index = Column('promptIndex', Integer, nullable=False, comment='提示词序号')
    title = Column('title', String(200), nullable=False, comment='提示词标题')
    content = Column('content', Text, nullable=False, comment='提示词内容')
    difficulty = Column('difficulty', String(20), nullable=True, comment='难度: easy/medium/hard')
    tags = Column('tags', String(500), nullable=True, comment='标签数组(JSON)')
    expected_output = Column('expectedOutput', Text, nullable=True, comment='期望输出(可选)')
    create_time = Column('createTime', DateTime, nullable=False, default=datetime.now, comment='创建时间')
    update_time = Column('updateTime', DateTime, nullable=False, default=datetime.now, onupdate=datetime.now, comment='更新时间')
    is_delete = Column('isDelete', Integer, nullable=False, default=0, comment='是否删除')

    def __repr__(self):
        return f"<ScenePrompt(id={self.id}, title={self.title})>"
